src/EnergyPlus/DataZoneEquipment.cc

```diff
@@ -382,180 +379,171 @@ void GetZoneEquipmentData(EnergyPlusData &state)
-                    ShowContinueError(state,
-                                      format("invalid {}=[{}].",
-                                             cNumericFields(nNumsInExtensible * ZoneEquipTypeIdx + nNumsBeforeExtensible + 2),
-                                             thisZoneEquipList.HeatingPriority(ZoneEquipTypeNum)));
-                    ShowContinueError(state, "equipment sequence must be > 0 and <= number of equipments in the list.");
-                    if (thisZoneEquipList.HeatingPriority(ZoneEquipTypeNum) > 0)
-                        ShowContinueError(state, format("only {} in the list.", thisZoneEquipList.NumOfEquipTypes));
-                    state.dataZoneEquip->GetZoneEquipmentDataErrorsFound = true;
-                }
+            std::string loadDistName = ip->getAlphaFieldValue(epListFields, objectSchemaProps, "load_distribution_scheme");
+            thisZoneEquipList.LoadDistScheme = static_cast<DataZoneEquipment::LoadDist>(
+                getEnumerationValue(DataZoneEquipment::LoadDistNamesUC, UtilityRoutines::MakeUPPERCase(loadDistName)));
+            if (thisZoneEquipList.LoadDistScheme == DataZoneEquipment::LoadDist::Invalid) {
+                ShowSevereError(state, format("{}{} = \"{}, Invalid choice\".", RoutineName, CurrentModuleObject, thisZoneEquipList.Name));
+                ShowContinueError(state, format("...load_distribution_scheme=\"{}\".", loadDistName));
+                state.dataZoneEquip->GetZoneEquipmentDataErrorsFound = true;
+            }
 
-                const int coolingFractionArrayIdx = nAlphasInExtensible * ZoneEquipTypeIdx + nAlphasBeforeExtensible + 3;
-                if (lAlphaBlanks(coolingFractionArrayIdx)) {
-                    thisZoneEquipList.SequentialCoolingFractionSchedPtr(ZoneEquipTypeNum) = DataGlobalConstants::ScheduleAlwaysOn;
-                } else {
-                    thisZoneEquipList.SequentialCoolingFractionSchedPtr(ZoneEquipTypeNum) =
-                        GetScheduleIndex(state, AlphArray(coolingFractionArrayIdx));
-                    if (thisZoneEquipList.SequentialCoolingFractionSchedPtr(ZoneEquipTypeNum) == 0) {
-                        ShowSevereError(state, std::string{RoutineName} + CurrentModuleObject + "=\"" + AlphArray(1) + "\".");
-                        ShowContinueError(state,
-                                          "invalid " + cAlphaFields(coolingFractionArrayIdx) + "=[" + AlphArray(coolingFractionArrayIdx) + "].");
-                        ShowContinueError(state, "Schedule does not exist.");
+            auto extensibles = epListFields.find("equipment");
+            if (extensibles != epListFields.end()) {
+                auto extensiblesArray = extensibles.value();
+                thisZoneEquipList.NumOfEquipTypes = static_cast<int>(extensiblesArray.size());
+
+                // Increment overall count of equipment
+                overallEquipCount += thisZoneEquipList.NumOfEquipTypes;
```

Comment:
Here's the fix. Instead of computing the number of extensible groups based on the entries in the AlphArrays, we use the epJSON extensibles, and the problem goes away

Reply:
I have run into this before where I had to add a schedule name. I thought this was a required field but now I think the problem only happened when there were more that 2 equipment types listed.

Reply:
Yes, you need 3 to get the original error.